core/example/tutorial/04_aosoa_cuda/aosoa_cuda.cpp

```diff
@@ -0,0 +1,219 @@
+/****************************************************************************
```

Comment:
I would say that this in fact demonstrates "portability" - least in the context of version 0.1. Because we have no parallel for concept in this release we will need to rely on the users to use their own programming model. I think these examples show the same code can be used with the different programming model (kokkos or otherwise).

What I could do is combine these into a single file and use `#ifdef CABANA_ENABLE_somebackend` the protect different implementations. Note sure exactly how I would implement the rest of that but I bet it is possible and would give the user a single file to look at.

Reply:
I personally think that one file could be better, but I'm happy to be flexible

Reply:
So it's not clear to my be looking at this how I would combine these into one file without making a mess. These examples demonstrate our portable data structures. It's hard to cleanly demonstrate portable parallelism without our `parallel_for`. Once we push that in version 0.2 I think then we will have a single clean example with `AoSoA`, `Slice`, etc. for the user to look at.  

What if we refactored these examples in version 0.2 once we have a `parallel_for`?

Reply:
@rfbird does this work for you?

Reply:
Sure sounds fine to me. I don't personally care, what worries me is that someone will see this, think they have to write device code and we'll never get a second chance to capture them as a user. More of a management concern that a developer one

Reply:
Maybe we could put some of the simple examples into one dir, which I believe can be useful for some people. The important ones will be those with parallel_for which may be in another dir.

Reply:
I'm starting to think should we even have Cuda/OpenMP variants at all at this point without our `parallel_for`? I do want users to know that they can use this data structure with different programming models but in ideal cases they shouldn't have to do much of anything with a different programming model when it comes to particles.

What I could do is just keep the serial example for now and explain in comments how to change the memory space in which things are allocated and that the data structures can be used with programming models compatible in those memory spaces.

Reply:
Ultimately I agree with Bob's point - we don't want potential users to think they need to use other programming models for their particles.

Reply:
OK - just pushed a commit to this effect